src/Compilers/CSharp/Portable/Symbols/Synthesized/PrivateImplementationDetailsLanguageProxy.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Roslyn.Utilities;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CodeGen;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    // A language-specific Symbol proxy that takes the place of the PrivateImplementationDetails type.
+    internal class PrivateImplementationDetailsLanguageProxy : NamedTypeSymbol
+    {
+        private readonly CSharpCompilation _compilation;
+        private readonly PrivateImplementationDetails _privateImplDetails;
+
+        public PrivateImplementationDetailsLanguageProxy(CSharpCompilation compilation, PrivateImplementationDetails privateImplDetails)
+        {
+            this._compilation = compilation;
+            this._privateImplDetails = privateImplDetails;
+        }
+
+        public override int Arity => 0;
+        public override NamedTypeSymbol ConstructedFrom => this;
+        public override Symbol ContainingSymbol => _compilation.GlobalNamespace;
```

Comment:
This type should belong to the source module, so the containing symbol should be that module's GlobalNamespace not the one from the compilation.